src/uu/ls/src/ls.rs

```diff
@@ -2702,8 +2702,7 @@ fn file_is_executable(md: &Metadata) -> bool {
     // S_IXUSR -> user has execute permission
     // S_IXGRP -> group has execute permission
     // S_IXOTH -> other users have execute permission
-    #[allow(clippy::unnecessary_cast)]
-    return md.mode() & ((S_IXUSR | S_IXGRP | S_IXOTH) as u32) != 0;
+    md.mode() & ((S_IXUSR | S_IXGRP | S_IXOTH) as u32) != 0
```

Comment:
please revert it
if I remember correctly, it is necessary for some archs

Reply:
Yep! reverted it. My clippy version was outdated, so it recognize that the return was necessary due to the attribute